daemon/cmd/daemon.go

```diff
@@ -941,6 +941,14 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		return nil, nil, err
 	}
 
+	// allocateIPs got us the routerIP so now we can create ipsec endpoint
+	// we must do this before publishing the router IP otherwise remote
+	// nodes could pick up the IP and send us outer headers we do not yet
+	// have xfrm rules for.
```

Comment:
nit: I'd probably add a comment to the effect of this to ensure later refactors can take into account the dependencies for this new logic:
```suggestion
	// have xfrm rules for.
	//
	// This must occur strictly after `setupIPSec()` and `d.allocateIPs()`.
```

Reply:
will do.